src/test/bls_tests.cpp

```diff
@@ -10,6 +10,17 @@
 
 BOOST_FIXTURE_TEST_SUITE(bls_tests, BasicTestingSetup)
 
+BOOST_AUTO_TEST_CASE(bls_sethexstr_tests)
+{
+    CBLSSecretKey sk;
+    BOOST_CHECK(sk.SetHexStr("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"));
+    BOOST_CHECK(!sk.SetHexStr("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e"));
+    BOOST_CHECK(!sk.SetHexStr("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1"));
+    BOOST_CHECK(!sk.SetHexStr("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1fzz"));
+    // invalid SetHexStr()-s should not alter internal data set via a valid one earlier
+    BOOST_CHECK(sk.ToString() == "000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f");
```

Comment:
I'd prefer `SetHexStr` actually invalidating internal data, so that `IsValid()` returns false after an invalid call. Reason is that otherwise security issues could arise if we re-use key objects (which we shouldn't do!) for some reason.